lib/oneboxer.rb

```diff
@@ -293,6 +294,25 @@ def self.local_user_html(url, route)
     end
   end
 
+  def self.local_category_html(url, route)
+    return unless route[:category_slug_path_with_id]
+    category = Category.find_by_slug_path_with_id(route[:category_slug_path_with_id])
+
+    if Guardian.new.can_see_category?(category)
```

Comment:
if category_slug_path_with_id is not found can users see it? perhaps double check for nil here?

Reply:
Looks like `can_see_category?` should take care of that for us:

https://github.com/discourse/discourse/blob/master/lib/guardian/category_guardian.rb#L50

Reply:
cool, ok merging this!